thunder/core/proxies.py

```diff
@@ -1212,13 +1212,15 @@ def _infer_tensor_properties(
     device: devices.Device | None = None,
     dtype: dtypes.dtype | None = None,
     requires_grad: bool | None = None,
+    grad: TensorProxy | None = None,
```

Comment:
I'm worried about having `grad` and `requires_grad` attributes on intermediate TensorProxies. Some meta function implementers may try setting these attributes but they are simply ignored by current forward-backward trace generation. An idea for the future is to have a separate TensorProxy subclass to represent input TensorProxies with `grad` and `requires_grad` attributes set.

Reply:
Indeed, the arguments of `TensorProxy.__init__` expose too much implementation details. It could be a good idea to provide a separate constructor or a parent class with less attributes for use in meta functions.